pkg/generators/models/models.go

```diff
@@ -253,6 +255,16 @@ func fillValidationRelatedProperties(ref *openapi3.SchemaRef, spec *PropSpec) (i
 		spec.NeedsValidation = true
 		spec.IsURL = true
 		imports = append(imports, "github.com/go-ozzo/ozzo-validation/v4/is")
+	case "uri":
+		spec.IsRequiredInValidation = !spec.IsNullable && spec.IsRequired
+		spec.NeedsValidation = true
+		spec.IsURI = true
+		imports = append(imports, "github.com/go-ozzo/ozzo-validation/v4/is")
```

Comment:
is there any reason not to use a map for this?

Reply:
:thinking: i guess it is slightly easier to render in the go template?

Reply:
@LucasRoesler then at some point you need to filter out all the duplicates, I'm not saying to use a map in templates, you can't do that if you want deterministic output but for filling this data it's way easier to collect it to a map and then export to a sorted slice.

Reply:
I this is what it would look like with a map

```diff
diff --git pkg/generators/models/models.go pkg/generators/models/models.go
index 0f5ade1..88d9ee3 100644
--- pkg/generators/models/models.go
+++ pkg/generators/models/models.go
@@ -161,8 +161,9 @@ func structPropsFromRef(ref *openapi3.SchemaRef) (specs []PropSpec, imports []st
 			IsNullable:  prop.Value.Nullable,
 		}
 
+		importIdx := map[string]struct{}{}
 		if spec.GoType == "time.Time" || spec.GoType == "*time.Time" {
-			imports = append(imports, "time")
+			importIdx["time"] = struct{}{}
 		}
 
 		spec.JSONTags = "`json:\"" + name
@@ -171,14 +172,21 @@ func structPropsFromRef(ref *openapi3.SchemaRef) (specs []PropSpec, imports []st
 		}
 		spec.JSONTags += "\"`"
 
-		imports = append(imports, fillValidationRelatedProperties(prop, &spec)...)
+		for name, value := range fillValidationRelatedProperties(prop, &spec) {
+			importIdx[name] = value
+		}
+
+		for name, _ := range importIdx {
+			imports = append(imports, name)
+		}
 
 		specs = append(specs, spec)
 	}
-	return specs, uniqueStrings(imports), err
+	return specs, imports, err
 }
 
-func fillValidationRelatedProperties(ref *openapi3.SchemaRef, spec *PropSpec) (imports []string) {
+func fillValidationRelatedProperties(ref *openapi3.SchemaRef, spec *PropSpec) (imports map[string]struct{}) {
+
 	if (len(spec.GoType) > 0 && spec.GoType[0] >= 'A' && spec.GoType[0] <= 'Z') ||
 		(len(spec.GoType) > 1 && spec.GoType[:2] == "[]") ||
 		(len(spec.GoType) > 2 && spec.GoType[:3] == "map") {
@@ -234,57 +242,57 @@ func fillValidationRelatedProperties(ref *openapi3.SchemaRef, spec *PropSpec) (i
 		spec.IsRequiredInValidation = !spec.IsNullable && spec.IsRequired
 		spec.IsRequired = true
 		spec.IsDateTime = true
-		imports = append(imports, "time")
+		imports["time"] = struct{}{}
 	case "byte":
 		spec.IsRequiredInValidation = !spec.IsNullable && spec.IsRequired
 		spec.NeedsValidation = true
 		spec.IsBase64 = true
-		imports = append(imports, "github.com/go-ozzo/ozzo-validation/v4/is")
+		imports["github.com/go-ozzo/ozzo-validation/v4/is"] = struct{}{}
 	case "email":
 		spec.IsRequiredInValidation = !spec.IsNullable && spec.IsRequired
 		spec.NeedsValidation = true
 		spec.IsEmail = true
-		imports = append(imports, "github.com/go-ozzo/ozzo-validation/v4/is")
+		imports["github.com/go-ozzo/ozzo-validation/v4/is"] = struct{}{}
 	case "uuid":
 		spec.IsRequiredInValidation = !spec.IsNullable && spec.IsRequired
 		spec.NeedsValidation = true
 		spec.IsUUID = true
-		imports = append(imports, "github.com/go-ozzo/ozzo-validation/v4/is")
+		imports["github.com/go-ozzo/ozzo-validation/v4/is"] = struct{}{}
 	case "url":
 		spec.IsRequiredInValidation = !spec.IsNullable && spec.IsRequired
 		spec.NeedsValidation = true
 		spec.IsURL = true
-		imports = append(imports, "github.com/go-ozzo/ozzo-validation/v4/is")
+		imports["github.com/go-ozzo/ozzo-validation/v4/is"] = struct{}{}
 	case "uri":
 		spec.IsRequiredInValidation = !spec.IsNullable && spec.IsRequired
 		spec.NeedsValidation = true
 		spec.IsURI = true
-		imports = append(imports, "github.com/go-ozzo/ozzo-validation/v4/is")
+		imports["github.com/go-ozzo/ozzo-validation/v4/is"] = struct{}{}
 	case "request-uri":
 		spec.IsRequiredInValidation = !spec.IsNullable && spec.IsRequired
 		spec.NeedsValidation = true
 		spec.IsRequestURI = true
-		imports = append(imports, "github.com/go-ozzo/ozzo-validation/v4/is")
+		imports["github.com/go-ozzo/ozzo-validation/v4/is"] = struct{}{}
 	case "hostname":
 		spec.IsRequiredInValidation = !spec.IsNullable && spec.IsRequired
 		spec.NeedsValidation = true
 		spec.IsHostname = true
-		imports = append(imports, "github.com/go-ozzo/ozzo-validation/v4/is")
+		imports["github.com/go-ozzo/ozzo-validation/v4/is"] = struct{}{}
 	case "ipv4":
 		spec.IsRequiredInValidation = !spec.IsNullable && spec.IsRequired
 		spec.NeedsValidation = true
 		spec.IsIPv4 = true
-		imports = append(imports, "github.com/go-ozzo/ozzo-validation/v4/is")
+		imports["github.com/go-ozzo/ozzo-validation/v4/is"] = struct{}{}
 	case "ipv6":
 		spec.IsRequiredInValidation = !spec.IsNullable && spec.IsRequired
 		spec.NeedsValidation = true
 		spec.IsIPv6 = true
-		imports = append(imports, "github.com/go-ozzo/ozzo-validation/v4/is")
+		imports["github.com/go-ozzo/ozzo-validation/v4/is"] = struct{}{}
 	case "ip":
 		spec.IsRequiredInValidation = !spec.IsNullable && spec.IsRequired
 		spec.NeedsValidation = true
 		spec.IsIP = true
-		imports = append(imports, "github.com/go-ozzo/ozzo-validation/v4/is")
+		imports["github.com/go-ozzo/ozzo-validation/v4/is"] = struct{}{}
 	case "", "int32", "int64", "float", "double":
 		break // do nothing
 	default:
```

Reply:
@pragmader just to keep the scope clean here and because this blocks one of my hub tickets, do you think we should make this change now or do a followup?

Reply:
@LucasRoesler up to you, it's more an optimization, there is nothing wrong with the current state of the PR.

Reply:
@LucasRoesler also, I believe it's not a full diff, you don't need the `uniqueStrings` function anymore, do you?

Reply:
you are right

Reply:
ok, i would rather do a second PR then